src/renderer/components/firmware/DownloadFirmwareButton.tsx

```diff
@@ -64,7 +74,52 @@ const DownloadFirmwareButton: React.FC<Props> = ({
   };
 
   const download = async (): Promise<void> => {
-    if (validPrVersion) {
+    // Try to get the cloudbuild build, if not found, throw an error
+    if (isCloudBuild && isCloudBuildValid) {
+      const flags = selectedFlags as { name: string; value: string }[];
+      const response = await client.query({
+        query: gql(/* GraphQL */ `
+          query CloudFirmwareStatus($params: CloudFirmwareParams!) {
+            cloudFirmwareStatus(params: $params) {
+              status
+              downloadUrl
+            }
+          }
+        `),
+        variables: {
+          params: {
+            release: version,
+            target,
+            flags,
+          },
+        },
+      });
+
+      const status = response.data.cloudFirmwareStatus.status as JobStatus;
+      console.log("STATUS", status);
+
+      const { downloadUrl } = response.data.cloudFirmwareStatus;
```

Comment:
If you have the URL, you can just download that here. I only did it through the backend for the github firmware as there was already complexity of unzipping the right firmware from a bundle etc

Reply:
Or see my comments above, you should query this by params

Reply:
Okay, changed it by directly using the downloadBinary method.

Reply:
Nah, don't do it this way. What's the reason not to just provide the data as a resolver on the job status?

IMO you should just make the structure of the resolver:
```
cloudFirmware(params: $params) {
  status
  base64data
}
```

Reply:
You definitely shouldn't be importing anything from the backend in the frontend btw. For types you should add those to `shared`

Reply:
Okay, fixed